### ISSUE TITLE
DOS and BS insertion to `mmdb.insert`

### ISSUE THREAD

**Issue by tschaume** [hackingmaterials/atomate#128]
The removal of DOS and BS from the task doc and its insertion to GridFS is currently implemented in `VaspToDb.run_task` (see [vasp.firetasks.parse_outputs](https://github.com/hackingmaterials/atomate/blob/d2742a3b076444f3093aec73d61a816496b3ca5e/atomate/vasp/firetasks/parse_outputs.py#L98-L114)). Would it make sense to move this part of the code into `mmdb.insert` to enable re-use when manually running drones (e.g. with `pymatgen.hive`)?

**Comment by matk86** [comment]
Insertion of bs and dos can be done using mmdb(object of VaspDb) without using the task. Instead of lumping the dos and bs insertion into the insert method separate methods are provided for bs and dos insertion i.e inserting dos and bs along the rest of the task doc is currently a 3 step process:
mmdb.insert_gridfs --> bs
mmdb.insert_gridfs --> dos
mmdb.insert --> the rest of the basic vasp output stuff

the insert method is inherited from the base class calcdb and the other 2 methods are defined only for vaspdb (subclasses​ calcdb). This is done so that the user can define their own inherited db class with customized vassp output data insertion , with the insert method only inserting the basic stuff by default...an example for such use case would be boltzrtrapdb that does some processing on the bandstructure before insertion(@computron and @shyamd would be able to comment more on this)...

**Comment by computron** [comment]
@tschaume not sure I understand what you are saying

The VaspDrone in pymatgen just returns a dict back (including band structure and DOS). I imagine this is the way a pymatgen user would want it. Note that there is no DB functionality (e.g., treatment of GridFS) as far as I can tell in pymatgen (not pymatgen-db). Let me know if I'm mistaken.

As for removing the dos/bs from that dict and putting it into GridFS rather than the main dict, it seems like something for either pymatgen-db or atomate to do. We can't put mmdb inside pymatgen-db since pymatgen-db doesn't depend on atomate. Anyway atomate does things differently than pymatgen-db and doesn't use its database insertion functionality.

Basically I think it's fine the way it is. If someone wants to just run the db insertion task outside of FireWorks, they can just do so. Just make the VaspToDb object and call run_task(). And if you want to do it in the context of "hive", you could probably just code your own Drone that does little more than create a VaspToDb Firetask and calls the run_task() method inside. But again that Drone would have an atomate dependency.

**Comment by computron** [comment]
Btw in the future I'm only answering tickets to:

https://groups.google.com/forum/#!forum/atomate

and closing others. Yes I know there are special features to Github issues like referencing commits, etc etc. But I prefer having a clear record of all conversations out in the open rather than having them disappear after closing them.

**Comment by computron** [comment]
@tschaume and I discussed more so I better understood what he was getting at. I refactored essentially according to his suggestion via 54d99a14da310378d858c35d4741c9d828161208

**Comment by tschaume** [comment]
Excellent! Thank you!
